src/cache/cache.rs

```diff
@@ -113,15 +125,12 @@ impl CacheRead {
     where
         T: Write,
     {
-        let file = self
-            .zip
-            .by_name(name)
-            .context("Failed to read object from cache entry")?;
+        let file = self.zip.by_name(name).or(Err(DecompressionFailure))?;
         if file.compression() != CompressionMethod::Stored {
-            bail!("Cache entry is not stored as a zstd blob");
+            bail!(DecompressionFailure);
         }
         let mode = file.unix_mode();
-        zstd::stream::copy_decode(file, to)?;
+        zstd::stream::copy_decode(file, to).or(Err(DecompressionFailure))?;
```

Comment:
It might be useful to differentiate between zipfile errors/corruption and zstd errors, probably for purposes of `std::fmt::Display`.  But that might be a distinction without a difference, dunno.

Reply:
Yeah, for my money, there's not a meaningful difference. Either way decompression failed and the course of action is going to be the same (namely, do the compilation and re-cache).